src/library/scala/Enumeration.scala

```diff
@@ -27,6 +27,11 @@ import scala.util.matching.Regex
  *  `Value` type member of the enumeration (`Value` selected on the stable
  *  identifier path of the enumeration instance).
  *
+ *  Values SHOULD NOT be added to an enumeration after its construction;
+ *  doing so makes the enumeration thread-unsafe. If values are added to an
+ *  enumeration from multiple threads (in a non-synchronized fashion) after
```

Comment:
> "if you do this, it must be synchronized externally."

One of the things I want to emphasize is that, even if you *do* synchronize it properly, you shouldn't add values to an enumeration after construction. It's more of an unintended behaviour rather than an intended feature. I think it's more that there isn't a way to prevent adding values after construction, more than that it's "allowed".